src/azure-stac-cli/stac/processors/core/ingest_stac_collection.py

```diff
@@ -0,0 +1,119 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from core.processor import BaseProcessor
+from stac.core.metrics import sendmetrics
+
+class StacCol2Postgres(BaseProcessor):
+    
+    TEMPLATE_NAME = 'Ingest STAC Collection'
+    VERSION = '1.0'
+    
+    # todo: this initialization may not be needed, test and remove it
+    PGHOST = ''
+    PGUSER = ''
+    PGDATABASE = ''
+    PGPASSWORD = ''
+    CONN_STRING = ''
+    CONTAINER_NAME = ''
+    
+    def __init__(self):
+        
+        super().__init__()
+        
+    def __get_envvars(self) -> None:
+        """Get all the environment variables relevant for this processor
+        :returns: None
+        :rtype: None
+        """
+        import os
+        
+        self.PGHOST = os.getenv('PGHOST')
+        self.PGUSER = os.getenv('PGUSER')
+        self.PGDATABASE = os.getenv('PGDATABASE')
+        self.PGPASSWORD = os.getenv('PGPASSWORD')
+        self.CONN_STRING = os.getenv('DATA_STORAGE_ACCOUNT_CONNECTION_STRING')
+        self.CONTAINER_NAME = os.getenv('STACCOLLECTION_STORAGE_CONTAINER_NAME')
+    
+    @sendmetrics
+    def run(self, **kwargs):
+        """ Ingest STAC collection to PostgreSQL """
+        
+        import os
+        import json
+        import asyncio
+        import psycopg2
```

Comment:
Looks like my changes from [here](https://github.com/Azure/Azure-Orbital-STAC/pull/26/files) got lost. Might be good to change this back to the newer psycopg 3 version, since that's already pulled in my pgstac anyway.

Reply:
Rebased the changes from the main branch.